kernel/src/vm/vmar/mod.rs

```diff
@@ -1384,4 +1385,143 @@ mod test {
+    }
+
+    #[ktest]
+    fn test_cow_copy_pt_iomem() {
+        let vm_space = VmSpace::new();
+        let map_range = PAGE_SIZE..(PAGE_SIZE * 2);
+        let cow_range = 0..PAGE_SIZE * 512 * 512;
+        let page_property = PageProperty::new_user(PageFlags::RW, CachePolicy::Uncacheable);
+        let preempt_guard = disable_preempt();
+
+        // Create and map IoMem instead of a frame.
+        let iomem_paddr = 0x50000; // Use a safe test physical address
+        let iomem = create_dummy_iomem(iomem_paddr, PAGE_SIZE);
+        let iomem_clone_for_assert = iomem.clone();
+
+        vm_space
+            .cursor_mut(&preempt_guard, &map_range)
+            .unwrap()
+            .map_iomem(iomem.clone(), page_property, PAGE_SIZE, 0);
+
+        // Confirms the initial mapping.
+        assert!(matches!(
+            vm_space.cursor(&preempt_guard, &map_range).unwrap().query().unwrap(),
+            (va, Some(VmQueriedItem::MappedIoMem { paddr, prop }))  if va.start == map_range.start && paddr == iomem_paddr && prop.flags == PageFlags::RW
+        ));
+
+        // Creates a child page table with copy-on-write protection.
+        let child_space = VmSpace::new();
+        {
+            let mut child_cursor = child_space.cursor_mut(&preempt_guard, &cow_range).unwrap();
+            let mut parent_cursor = vm_space.cursor_mut(&preempt_guard, &cow_range).unwrap();
+            let num_copied = cow_copy_pt(&mut parent_cursor, &mut child_cursor, cow_range.len());
+            assert_eq!(num_copied, 0); // IoMem pages are not "copied" in the same sense as RAM
+        };
+
+        // Confirms that parent and child VAs map to the same physical address.
+        {
+            let child_map_paddr = {
+                let (_, Some(VmQueriedItem::MappedIoMem { paddr, .. })) = child_space
+                    .cursor(&preempt_guard, &map_range)
+                    .unwrap()
+                    .query()
+                    .unwrap()
+                else {
+                    panic!("Child mapping query failed");
+                };
+                paddr
+            };
+            let parent_map_paddr = {
+                let (_, Some(VmQueriedItem::MappedIoMem { paddr, .. })) = vm_space
+                    .cursor(&preempt_guard, &map_range)
+                    .unwrap()
+                    .query()
+                    .unwrap()
+                else {
+                    panic!("Parent mapping query failed");
+                };
+                paddr
+            };
+            assert_eq!(child_map_paddr, parent_map_paddr);
+            assert_eq!(child_map_paddr, iomem_paddr);
+        }
+
+        // Unmaps the range from the parent.
+        vm_space
+            .cursor_mut(&preempt_guard, &map_range)
+            .unwrap()
+            .unmap(map_range.len());
+
+        // Confirms that the child VA remains mapped (IoMem mappings should persist).
```

Comment:
No child pages should be affected by unmapping pages in the parent page table, which is not related to whether the pages are `IoMem` pages or not.
```suggestion
        // Confirms that the child VA remains mapped.
```